### PR TITLE
[remix] Don't depend on `@remix-run/vercel`

### PR DESCRIPTION
**Pull request by TooTallNate**
Instead, just add it to the project's `package.json` file before installing the dependencies.

Fixes warning about missing peer dependencies when installing CLI.

CleanShot 2022-05-22 at 09 40 09@2x

Now, a warning is shown in the Deployment build logs, saying that the dep was added, but that the user should commit the change:

Screen Shot 2022-06-27 at 8 15 19 PM

### CODE REVIEW

File: packages/remix/src/build.ts [hunk 1]
```diff
@@ -70,6 +78,52 @@ export const build: BuildV2 = async ({
     }
   }
 
+  // Ensure `@remix-run/vercel` is in the project's `package.json`
+  const packageJsonPath = await walkParentDirs({
+    base: repoRootPath,
+    start: workPath,
+    filename: 'package.json',
+  });
+  console.log({ packageJsonPath });
```
Comment (styfle) [inline packages/remix/src/build.ts]: Remove this log?

File: packages/remix/src/build.ts [hunk 2]
```diff
@@ -70,6 +78,52 @@ export const build: BuildV2 = async ({
...
+    let modified = false;
+    if (REMIX_RUNTIME_ADAPTER_NAME in devDependencies) {
+      dependencies[REMIX_RUNTIME_ADAPTER_NAME] =
+        devDependencies[REMIX_RUNTIME_ADAPTER_NAME];
+      delete devDependencies[REMIX_RUNTIME_ADAPTER_NAME];
+      console.log(
+        `Warning: Moving "${REMIX_RUNTIME_ADAPTER_NAME}" from \`devDependencies\` to \`dependencies\`. You should commit this change.`
+      );
+      modified = true;
+    } else if (!('@remix-run/vercel' in dependencies)) {
```
Comment (styfle) [inline packages/remix/src/build.ts]: Should this use the constant too?

File: packages/remix/src/build.ts [hunk 3]
```diff
@@ -70,6 +78,52 @@ export const build: BuildV2 = async ({
...
+      const packageJsonString = JSON.stringify(
+        {
+          ...packageJson,
+          dependencies,
+          devDependencies,
+        },
+        null,
+        2
+      );
+      await fs.writeFile(packageJsonPath, `${packageJsonString}\n`);
```
Comment (styfle) [inline packages/remix/src/build.ts:120]: Will this break lock files?

Comment (TooTallNate) [reply packages/remix/src/build.ts:120]: Shouldn't. `npm` updates the `package-lock.json` file after this update, at least.

File: packages/remix/package.json [hunk 4]
```diff
@@ -21,7 +21,6 @@
     "default-server.js"
   ],
   "dependencies": {
-    "@remix-run/vercel": "1.4.3",
     "@vercel/nft": "0.19.1"
```
Comment (styfle) [inline packages/remix/package.json:24]: Should we move all deps to devDeps and run ncc like the other builders?

**Review by styfle — APPROVED** [review]

**Pull request merged**